client/driver/docker.go

```diff
@@ -1458,10 +1463,48 @@ func calculatePercent(newSample, oldSample, newTotal, oldTotal uint64, cores int
 	return (float64(numerator) / float64(denom)) * float64(cores) * 100.0
 }
 
+func authOptionFromDriverConfig(auth DockerDriverAuth) (*docker.AuthConfiguration, error) {
+	return &docker.AuthConfiguration{
+		Username:      auth.Username,
+		Password:      auth.Password,
+		Email:         auth.Email,
+		ServerAddress: auth.ServerAddress,
+	}, nil
+}
+
+func authOptionFromHelper(helperName, repoName string, clientConfig *config.Config) (*docker.AuthConfiguration, error) {
+	helper := dockerAuthHelperPrefix + helperName
+	cmd := exec.Command(helper, "get")
+	cmd.Stdin = strings.NewReader(repoName)
+
+	// For helper to be able to read credentials from vault
+	cmd.Env = append(os.Environ(), fmt.Sprintf("VAULT_TOKEN=%s", clientConfig.VaultConfig.Token))
```

Comment:
I see your point! Once we have generic plugin support, a overall whitelist for the things Nomad fork/execs to delegate work would probably make sense. I can see the Vault use case being nice for people writing custom credential-helpers. However I would like even the first cut to allow operators to control which ones have access, so given it expands the PR I would be okay if you want to skip that functionality for the first PR!